service/history/interfaces/shard_context.go

```diff
@@ -120,6 +121,9 @@ type (
 		// If branchToken != nil, then delete history also, otherwise leave history.
 		DeleteWorkflowExecution(ctx context.Context, workflowKey definition.WorkflowKey, branchToken []byte, closeExecutionVisibilityTaskID int64, workflowCloseTime time.Time, stage *tasks.DeleteWorkflowExecutionStage) error
 
+		GetCachedWorkflowContext(ctx context.Context, namespaceID namespace.ID, execution *commonpb.WorkflowExecution, lockPriority locks.Priority) (WorkflowContext, ReleaseWorkflowContextFunc, error)
+		GetCurrentCachedWorkflowContext(ctx context.Context, namespaceID namespace.ID, workflowID string, lockPriority locks.Priority) (ReleaseWorkflowContextFunc, error)
```

Comment:
I still don't view Cache as an implementation detail of shard context which is persistence facing, and think Cache is one layer on top of that. 

I view those two methods as providing syntax sugar for doing `shardContext.GetWorkflowCache().GetWorkflowContext()` and using shard context as property bag for providing access to workflow cache (a host level component) thus saving us from always passing both shard context and workflow cache around to all components.

That also means if later we agreed on an approach for not using shard context as the property bag, those two methods should be removed as well. 

My approval is based on ^ interpretation of those two methods.